src/eigen/lobpcg_hyper_impl.jl

```diff
@@ -310,17 +310,17 @@ end
     X
 end
 
-
 function final_retval(X, AX, BX, λ, resid_history, niter, n_matvec)
-    if !issorted(λ)
-        p = sortperm(λ)
-        λ = λ[p]
+    λ_host = oftype(ones(eltype(λ), 1), λ) # Copy to CPU for element-wise access
```

Comment:
Since you are not copying it back to the device: Is this what we want ? Is this maybe what we do anyway in diag.jl (that is putting the eigenvalues on the cpu) ?

Reply:
Alternative is maybe to return \lambda[sortperm] ?

Reply:
There was actually a mistake, in which the unsorted eigenvalues where returned. This is fixed now (but was not caught by the tests).

As for not copying back to the GPU, this is indeed consistent with `diag.jl`, where the eigenvalues are transferred to the host anyways. So doing it at this stage isn't a performance overhead, and it is what we want.

Reply:
Can you just add an appropriate note in the LOBPCG docstring? To me this would otherwise come as a surprise.